core/src/main/scala/io/chrisdavenport/shellfish/Shell.scala

```diff
@@ -383,20 +395,24 @@ object Shell {
         .flatMap(list =>
           Stream
             .emits(list)
-            .map(Paths.get(_))
-            .flatMap(p => files.walk(p, 1))
-            .filter(f =>
-              f.getFileName.toString == path && Files.isExecutable(f)
+            .map(Path(_))
+            .flatMap(singleWalk)
+            .flatMap(p =>
+              Stream
+                .eval(files.isExecutable(p))
+                .map(isExe => (p, isExe && p.fileName == Path(path)))
             )
-            .map(_.toString)
+            .filter(_._2)
+            .map(_._1.toString)
```

Comment:
```suggestion
            .collect {
              case x if x._2 => x._1.toString
            }
```
`filter.map == collect`